common-http/src/main/java/io/cdap/common/http/HttpRequests.java

```diff
@@ -103,31 +102,45 @@ public static HttpResponse execute(HttpRequest request, HttpRequestConfig reques
 
     conn.connect();
 
-    try {
-      if (bodySrc != null) {
-        try (InputStream input = bodySrc.getInput(); OutputStream os = conn.getOutputStream()) {
-          ByteStreams.copy(input, os);
-        }
-      }
 
-      try {
-        if (isSuccessful(conn.getResponseCode())) {
-          try (InputStream inputStream = conn.getInputStream()) {
-            return new HttpResponse(conn.getResponseCode(), conn.getResponseMessage(),
-                                    ByteStreams.toByteArray(inputStream), conn.getHeaderFields());
-          }
-        }
-      } catch (FileNotFoundException e) {
-        // Server returns 404. Hence handle as error flow below. Intentional having empty catch block.
+    if (bodySrc != null) {
```

Comment:
This is a very long method. Can you refactor it with some private method to make the logic cleaner?

Reply:
refactored